napari/layers/_shapes_layer/model.py

```diff
@@ -510,41 +573,59 @@ def add_shapes(self, data, *, shape_type='rectangle', edge_width=1,
             return
 
         if np.array(data[0]).ndim == 1:
-            # If a single array for a shape has been passed
-            if shape_type in self.data._types.keys():
-                shape_cls = self.data._types[shape_type]
-                shape = shape_cls(data, edge_width=edge_width,
-                                  edge_color=edge_color, face_color=face_color,
-                                  opacity=opacity, z_index=z_index)
-            else:
-                raise ValueError("""shape_type not recognized, must be one of
-                                 "{'line', 'rectangle', 'ellipse', 'path',
-                                 'polygon'}"
-                                 """)
-            self.data.add(shape)
-        else:
-            # Turn input arguments into iterables
-            shape_types = ensure_iterable(shape_type)
-            edge_widths = ensure_iterable(edge_width)
-            opacities = ensure_iterable(opacity)
-            z_indices = ensure_iterable(z_index)
-            edge_colors = ensure_iterable(edge_color, color=True)
-            face_colors = ensure_iterable(face_color, color=True)
-
-            for d, st, ew, ec, fc, o, z, in zip(data, shape_types, edge_widths,
-                                                edge_colors, face_colors,
-                                                opacities, z_indices):
-                shape_cls = self.data._types[st]
-                shape = shape_cls(d, edge_width=ew, edge_color=ec,
+            # If a single array for a shape has been passed turn into list
+            data = [data]
+
+        # Turn input arguments into iterables
+        shape_types = ensure_iterable(shape_type)
+        edge_widths = ensure_iterable(edge_width)
+        opacities = ensure_iterable(opacity)
+        z_indices = ensure_iterable(z_index)
+        edge_colors = ensure_iterable(edge_color, color=True)
+        face_colors = ensure_iterable(face_color, color=True)
+
+        for d, st, ew, ec, fc, o, z, in zip(data, shape_types, edge_widths,
+                                            edge_colors, face_colors,
+                                            opacities, z_indices):
+            shape_cls = self._data_view._types[st]
+
+            # Slice data by 2D plane.
+            key, data_2D = slice_by_plane(d)
+            if key is not False:
```

Comment:
Maybe use `None` as the value for no keys?

Reply:
I'm also unsure what `key` means here. Maybe rename to `slice_key`? Then you could say:

```python
if slice_key is not None:  # None implies broadcast mode
```

Reply:
(is that what it implies? I'm not sure.)

Reply:
renamed `slice_key` and added a note about `False` being invalid